sympy/solvers/diophantine.py

```diff
@@ -232,21 +234,30 @@ def classify_diop(eq):
     ([x, y], {1: 5, x: 1, x**2: 1, y: 0, y**2: 1, x*y: -1}, 'binary_quadratic')
     """
     eq = eq.expand(force=True)
-    var = list(eq.free_symbols)
-    var.sort(key=default_sort_key)
 
-    coeff = {}
+    coeff = coeff = eq.as_coefficients_dict()
```

Comment:
Why `coeff = coeff = eq.as_coefficients_dict()` instead of `coeff = eq.as_coefficients_dict()`?